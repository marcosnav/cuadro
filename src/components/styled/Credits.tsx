import styled from 'styled-components';

interface IProps {
  bottom?: boolean;
}

const Credits = styled.p<IProps>`
  bottom: ${({ bottom }) => bottom ? '0' : 'auto'};
  color: ${(props) => props.theme.TEXT};
  font-size: 0.7em;
  left: ${({ bottom }) => bottom ? '0' : 'auto'};
  line-height: 1.44em;
  position: ${({ bottom }) => bottom ? 'fixed' : 'relative'};
  text-align: center;
  width: 100%;
`;

Credits.displayName = 'Credits';

export default Credits;
