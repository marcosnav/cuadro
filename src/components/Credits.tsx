import React, { FC } from 'react';
import PrettyLink from './PrettyLink';
import * as S from './styled';

const Credits: FC = () => {
  return (
    <S.Text bottom={true}>
      {'Made with </> by '}
      <PrettyLink href='https://github.com/marcosnav/cuadro'>
        marcosnav.
      </PrettyLink>
      {' See project in '}
      <PrettyLink href='https://github.com/marcosnav/cuadro'>
        GitHub.
      </PrettyLink>
    </S.Text>
  );
};

export default Credits;
