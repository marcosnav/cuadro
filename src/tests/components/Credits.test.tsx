import { shallow } from 'enzyme';
import React from 'react';
import Credits from './../../components/Credits';

describe('<Credits/>', () => {
  test('renders properly', () => {
    const component = shallow(<Credits />);
    expect(component).toMatchSnapshot();
  });
});
