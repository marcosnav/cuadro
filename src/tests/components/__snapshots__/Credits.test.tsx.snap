// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Credits/> renders properly 1`] = `
<Text
  bottom={true}
>
  Made with &lt;/&gt; by 
  <PrettyLink
    href="https://github.com/marcosnav/cuadro"
  >
    marcosnav.
  </PrettyLink>
   See project in 
  <PrettyLink
    href="https://github.com/marcosnav/cuadro"
  >
    GitHub.
  </PrettyLink>
</Text>
`;
